core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -192,6 +211,7 @@ oppia.factory('CollectionObjectFactory', [
     // skills from all collections nodes within this domain object. Please note
     // this operation returns a new skill list everytime this function is
     // called.
+    // TODO(wxy): migrate this to use the skills dict instead.
```

Comment:
But it's not quite right anymore, right? ``SkillListObjectFactory`` stores a list of skills, but now we're storing IDs.